### ISSUE TITLE
Empty result

### ISSUE THREAD

**Issue by meijer3** [stefandreyer/JSON-Library#3]
Hi,
I got an empty string back
![image](https://user-images.githubusercontent.com/45625341/210442029-887d71d7-f146-4cd9-a0c1-4e70ab9e4f2d.png)

<pre>
TYPE MQTT_DEVICE :
STRUCT
	name: JSONVAR;
	configuration_url: JSONVAR;
	// connections: STRING(500); 
	identifiers:JSONVAR;
	manufacturer: JSONVAR;
	model:JSONVAR;
	sw_version: JSONVAR;
END_STRUCT
END_TYPE

---
LocalJSONObj.name.CharString := 'Wago';
LocalJSONObj.configuration_url.CharString := 'http//wago.ii';
LocalJSONObj.identifiers.CharString := 'Wago-750-8207-000-1';
LocalJSONObj.manufacturer.CharString := 'Wago';
LocalJSONObj.model.CharString := '750-8207-000';
LocalJSONObj.sw_version.CharString := '1.0';
</pre>

**Comment by stefandreyer** [comment]
Hi,

seems like you renamed your application like in:

https://github.com/stefandreyer/JSON-Library/issues/2

[PRO_JSON 1.0.0.15.zip](https://github.com/stefandreyer/JSON-Library/files/10349388/PRO_JSON.1.0.0.15.zip)

You may undone this or use the attached lib, where you can spezify the application name.

BR Stefan

**Comment by meijer3** [comment]
I did both, but none are working.
I see that the `NumberOfVars` is correct for example.
However the `names[0].Name` = '' 
The Error is `true` and ErrorPosition  = 0.

**Comment by stefandreyer** [comment]
Hi,

ApplicationLevel = 0 proves that you application name is not 'Application'.

It must be exactly 'Application' or the nam of the GVLP in the lib.

If so ApplicationLevel should be > 1.

Else, you may have a test project? 

Which PLC do you running? Same behaviour in simulation?

BR Stefan

**Comment by meijer3** [comment]
Sorry I am not super experienced with plc's.
The goal did you already discover:
https://github.com/MichielVanwelsenaere/HomeAutomation.CoDeSys3/issues/77

So I have a functionblok with a method. In that method I need to send MQTT to at topic:
The json looks something like (shortend): 
<pre>
{
  "~": "Devices/PLC/House",
  name: `Desk light`,
  device_class: "light",
  ...
  availability_template: "online", 
  device: {
      name: "Wago",
      configuration_url: "http://wago.ii",
      connections: [
        ["ip", "10.1.1.3"],
        ["mac", "yes:yes"],
      ],
      identifiers: "Wago-750-8207-000-1",
      manufacturer: "Wago",
      ...
   }
}
</pre>
I think this is the max levels. I am also curious about the first key: `~`. 

I run on a Wago PFC200 (7250-8207/025-000) with écockpit

**Comment by meijer3** [comment]
![image](https://user-images.githubusercontent.com/45625341/210960239-743b3148-ae01-474a-a483-649c90c21083.png)

**Comment by stefandreyer** [comment]
Hi,

please show me VarName in your JsonVar:
![image](https://user-images.githubusercontent.com/41088808/211066632-ed663182-4ec9-4f7b-abd4-038ee6608e70.png)

BR Stefan

**Comment by meijer3** [comment]
![image](https://user-images.githubusercontent.com/45625341/211133585-b58e02c9-f561-4872-9906-26ae63e2fa75.png)

**Comment by stefandreyer** [comment]
Hi,

this is bad, are all the VarName empty(other vars)?

If so, reflection doesn't work on wago, this writes the var path in VarName:
![image](https://user-images.githubusercontent.com/41088808/211163653-9eca6fa4-a016-4fa3-9fc4-7e28db836360.png)

can you check this with a own Var?

BR Stefan

**Comment by meijer3** [comment]
They all are empty. I worked in a featureblocks method. There refelection did not work. I moved to a POU, and there I do see the var path.
Device.Application.POU.OBJ.var
![image](https://user-images.githubusercontent.com/45625341/211166675-b460e349-75df-429f-803c-5c273f07d686.png)
But CharString is there = `''`

I loaded the default example....
![image](https://user-images.githubusercontent.com/45625341/211166723-4325f084-3b7a-4119-8edc-6ad20f2e64a2.png)

Thanks for spending your time in this!

**Comment by stefandreyer** [comment]
Hi,

if you running the example var1.boolean should be true, sure you calling the example?

Next, did you set ComposeJSON.Execute to true?

BR Stefan

**Comment by meijer3** [comment]
Awesome it does work :) (ofcourse no doubt there)
In a POU it works for now. With the example data, and ComposeJSON.Execute (altho its not in the example online).
I needed to download the whole program for it to work eventually. (thanks for pointing out var1 should be true)

In my functionblock not so much. 
In the debug mode, it shows for e.g. 
`obj.icon.VarName` `STRING(GPL_JSON.MAX_VAR_NAME)` = `''`

And 

`obj.icon.Names.Names[1].Name` `STRING(GPL_JSON.MAX_NAME_SIZE)` = `''`

**Comment by stefandreyer** [comment]
Hi,

this all works fine on my Raspberry. Did you tried to place your function Block in the POU where it's working?

I allways download the whole programm, maybe thats why I don't experence this behaviour.

BR Stefan

**Comment by stefandreyer** [comment]
One more:
![image](https://user-images.githubusercontent.com/41088808/211190708-3e5cf300-30fe-47f4-90ae-6f1d0b81d39d.png)

**Comment by MichielVanwelsenaere** [comment]
Not sure if it helps but I've been able to get reflection to work in my function blocks in éCockpit: 

![image](https://user-images.githubusercontent.com/6965657/211208205-40ce80f5-fcc6-4816-a7de-897e568395e7.png)

Note the 'reflection' piece of code above the FB name.

**Comment by stefandreyer** [comment]
Hi,

If this works for Michiel, then it have to for JSON.

BR Stefan

**Comment by meijer3** [comment]
Hi, 
Thanks both for pushing through. It works now in a FunctionBlock.
What I did before is that I tried to ComposeJSON.Execute inside a method in my FunctionBlock.
What I do now:
I set all variables inside the method, and save a`obj` in the `THIS^.obj` and execute in the FunctionBlock.
![image](https://user-images.githubusercontent.com/45625341/211304909-02c5ce92-4bfb-430e-9540-f39daf0ffaa8.png)
I still have the problem that MQTT only sends 128chars of the json.
